mungegithub/mungers/approval-handler.go

```diff
@@ -103,42 +104,32 @@ func (h *ApprovalHandler) Munge(obj *github.MungeObject) {
 
 		if len(fields) == 1 && strings.ToLower(fields[0]) == "/approve" {
 			approverSet.Insert(*c.User.Login)
-			continue
-		}
-
-		if len(fields) == 2 && strings.ToLower(fields[0]) == "/approve" && strings.ToLower(fields[1]) == "cancel" {
-			approverSet.Delete(*c.User.Login)
+		} else if len(fields) == 2 && strings.ToLower(fields[0]) == "/approve" && strings.ToLower(fields[1]) == "cancel" {
+			if approverSet.Has(*c.User.Login) {
+				approverSet.Delete(*c.User.Login)
+			}
 		}
 	}
 
+	//Checks that no valid approver has commented to cancel the approve label since last commit
+	//Checks that all files have an approver in the approverSet
 	for _, file := range files {
```

Comment:
New function: `AreAllFilesApproved() bool`.

(You're mixing the approval checking logic with the action that it implies)